how-to/workspace-platform-starter/client/src/modules/composite/default-workspace/default-workspace-storage.ts

```diff
@@ -0,0 +1,147 @@
+import type { EndpointClient } from "workspace-platform-starter/shapes/endpoint-shapes";
+import type { Logger } from "workspace-platform-starter/shapes/logger-shapes";
+import type { ModuleHelpers } from "workspace-platform-starter/shapes/module-shapes";
+import type { VersionInfo } from "workspace-platform-starter/shapes/version-shapes";
+import { isEmpty } from "workspace-platform-starter/utils";
+import type {
+	DefaultWorkspacePayload,
+	DefaultWorkspaceProviderOptions,
+	EndpointDefaultWorkspaceGetRequest,
+	EndpointDefaultWorkspaceGetResponse,
+	EndpointDefaultWorkspaceSetRequest
+} from "./shapes";
+
+/**
+ * A class that contains the methods required for saving and getting a default workspace.
+ */
+export class DefaultWorkspaceStorage {
+	/**
+	 * The logger for displaying information from the module.
+	 * @internal
+	 */
+	private readonly _logger?: Logger;
+
+	/**
+	 * Helper methods for the module.
+	 * @internal
+	 */
+	private readonly _helpers: ModuleHelpers | undefined;
+
+	/**
+	 * The helper methods to use.
+	 * @internal
+	 */
+	private readonly _settings?: DefaultWorkspaceProviderOptions;
+
+	/**
+	 * An endpoint client if available.
+	 * @internal
+	 */
+	private _endpointClient: EndpointClient | undefined;
+
+	/**
+	 * The version info for the currently running platform.
+	 * @internal
+	 */
+	private _versionInfo: VersionInfo | undefined;
+
+	/**
+	 * A helper for saving and returning the default workspace related information.
+	 * @param settings settings to be used by this helper
+	 * @param helpers helper functions to be used
+	 * @param logger a logger to use while performing actions
+	 */
+	constructor(
+		settings: DefaultWorkspaceProviderOptions | undefined,
+		helpers: ModuleHelpers | undefined,
+		logger: Logger
+	) {
+		this._logger = logger;
+		this._helpers = helpers;
+		this._settings = settings;
+		this.setupEndpointClient()
```

Comment:
Not a fan of using async methods in constructors, can they be moved to an init method

Reply:
done